flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/TestUtils.java

```diff
@@ -72,19 +75,26 @@ public static FlinkDeployment buildApplicationCluster() {
                         JobSpec.builder()
                                 .jarURI(SAMPLE_JAR)
                                 .parallelism(1)
+                                .upgradeMode(UpgradeMode.STATELESS)
                                 .state(JobState.RUNNING)
                                 .build());
         return deployment;
     }
 
     public static FlinkDeploymentSpec getTestFlinkDeploymentSpec() {
+        Map<String, String> conf = new HashMap<>();
+        conf.put(TaskManagerOptions.NUM_TASK_SLOTS.key(), "2");
+        conf.put(
```

Comment:
We could use `HighAvailabilityOptions.HA_MODE` and `KubernetesHaServicesFactory.class.getCanonicalName()` here.